src/thunderbots/software/test/network_input/filter/ball_filter.cpp

```diff
@@ -0,0 +1,348 @@
+    // When the ball is sitting still, the velocity could be any direction so we do
+    // not use a strict tolerance for this test. We only really care about the
+    // magnitude of the velocity to make sure it's small enough
+    Angle expected_velocity_angle_tolernace = Angle::threeQuarter();
+    double expected_velocity_magnitude_tolerance = 1.0;
+    int num_simulation_steps = 10;
+    int num_steps_to_ignore = 5;
+    Timestamp start_time = current_timestamp;
+
+    testFilterAlongBallTrajectory(start_time, ball_trajectory, ball_velocity_magnitude, ball_position_variance, time_step_variance, expected_position_tolerance, expected_velocity_angle_tolernace, expected_velocity_magnitude_tolerance, num_simulation_steps, num_steps_to_ignore);
+}
+
+TEST_F(BallFilterTest, ball_moving_slow_in_a_straight_line_with_no_noise_in_data) {
+    Segment ball_path = Segment(field.friendlyCornerNeg(), field.enemyCornerPos());
+    double ball_velocity_magnitude = 0.31;
+    double ball_position_variance = 0;
+    double time_step_variance = 0;
+    double expected_position_tolerance = 0.001;
+    // When the ball is sitting still, the velocity could be any direction so we do
+    // not use a strict tolerance for this test. We only really care about the
+    // magnitude of the velocity to make sure it's small enough
+    Angle expected_velocity_angle_tolernace = Angle::ofDegrees(0.01);
+    double expected_velocity_magnitude_tolerance = 0.01;
+    int num_steps_to_ignore = 5;
+    Timestamp start_time = current_timestamp;
+
+    testFilterAlongLineSegment(start_time, ball_path, ball_velocity_magnitude, ball_position_variance, time_step_variance, expected_position_tolerance, expected_velocity_angle_tolernace, expected_velocity_magnitude_tolerance, num_steps_to_ignore);
+}
+
+TEST_F(BallFilterTest, ball_moving_slow_in_a_straight_line_with_small_noise_in_data) {
+    Segment ball_path = Segment(field.friendlyCornerNeg(), field.enemyCornerPos());
+    double ball_velocity_magnitude = 0.3;
+    double ball_position_variance = 0.001;
+    double time_step_variance = 0.001;
+    double expected_position_tolerance = 0.005;
+    // When the ball is sitting still, the velocity could be any direction so we do
+    // not use a strict tolerance for this test. We only really care about the
+    // magnitude of the velocity to make sure it's small enough
+    Angle expected_velocity_angle_tolernace = Angle::ofDegrees(31);
+    double expected_velocity_magnitude_tolerance = 0.15;
+    int num_steps_to_ignore = 5;
+    Timestamp start_time = current_timestamp;
+
+    testFilterAlongLineSegment(start_time, ball_path, ball_velocity_magnitude, ball_position_variance, time_step_variance, expected_position_tolerance, expected_velocity_angle_tolernace, expected_velocity_magnitude_tolerance, num_steps_to_ignore);
+}
+
+// TODO: Enhance the filter and re-enable this test after resolving the following issue
+// https://github.com/UBC-Thunderbots/Software/issues/634
```

Comment:
Is this still an issue after all your recent changes?

Reply:
no

Reply:
no